ui/desktop/app/routes/scopes/scope/projects/targets/target.js

```diff
@@ -156,14 +162,35 @@ export default class ScopesScopeProjectsTargetsTargetRoute extends Route {
       const connectionDetails = await this.ipc.invoke('connect', options);
 
       // Associate the connection details with the session
+      let session;
       const { session_id, address, port, credentials } = connectionDetails;
-      const session = await this.store.findRecord('session', session_id);
+      try {
+        session = await this.store.findRecord('session', session_id);
+      } catch (error) {
+        /**
+         * if the user cannot read or fetch the session we add the important
+         * information returned from the connect command to allow the user
+         * to still continue their work with the information they need
+         */
+        this.store.pushPayload({
```

Comment:
I think we need to specify the type in this call otherwise it will only use the application serializer (even though we don't have any normalization specific for sessions).